homeassistant/components/thermostat/eq3btsmart.py

```diff
@@ -0,0 +1,206 @@
+"""
+Support for eq3 Bluetooth Smart thermostats.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/thermostat.max/
+"""
+import logging
+import struct
+from datetime import datetime
+
+from homeassistant.components.thermostat import ThermostatDevice
+from homeassistant.const import TEMP_CELCIUS
+from homeassistant.helpers.temperature import convert
+
+REQUIREMENTS = ['bluepy>=1.0.0']
+
+CONF_MAC = 'mac'
+CONF_DEVICES = 'devices'
+CONF_ID = 'id'
+
+PROP_WRITE_HANDLE = 0x411
+PROP_NTFY_HANDLE = 0x421
+PROP_ID_VALUE_PACKED = struct.pack('B', 0)
+PROP_GETINFO_VALUE_PACKED = struct.pack('B', 3)
+PROP_TEMPERATURE_VALUE = 0x41
+PROP_TEMPERATURE_VALUE_PACKED = struct.pack(
+    'B', PROP_TEMPERATURE_VALUE)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the eq3 BLE thermostats."""
+    devices = []
+
+    for name, device_cfg in config[CONF_DEVICES].items():
+        mac = device_cfg[CONF_MAC]
+        devices.append(EQ3BTSmartThermostat(mac, name))
+
+    add_devices(devices)
+    return True
+
+
+# pylint: disable=too-many-instance-attributes
+class EQ3BTSmartThermostat(ThermostatDevice):
+    """Representation of a Homematic thermostat."""
+
+    def __init__(self, _mac, _name):
+        """Initialize the thermostat."""
+        from bluepy import btle
+
+        self._mac = _mac
+        self._name = _name
+        self._target_temperature = -1
+        self._mode = -1
+        self._mode_readable = ''
+        self._request_value = -1
+        self._request_handle = -1
+
+        self._conn = btle.Peripheral()
+        self._connect()
+        self.update()
+
+    def get_delegate(self):
```

Comment:
You can move this method to the root of the file, that way it won't clutter the class definition.